cmd/benchttp/run.go

```diff
@@ -136,17 +138,25 @@ func (*cmdRun) requesterConfig(cfg config.Global) requester.Config {
 
 // handleRunInterrupt handles the case when the runner is interrupted.
 func (*cmdRun) handleRunInterrupt() error {
-	reader := bufio.NewReader(os.Stdin)
-	// TODO: list output strategies
-	// TODO: do not prompt if strategy is stdout only
-	// TODO: add config option "output.generateOnCancel" and remove prompt?
-	fmt.Printf("\nBenchmark interrupted, generate output anyway? (yes/no): ")
-	line, _, err := reader.ReadLine()
+	v, err := promptf("\nBenchmark interrupted, generate output anyway? (yes/no): ")
 	if err != nil {
 		return err
 	}
-	if string(line) != "yes" {
+	if v != "yes" {
 		return errors.New("benchmark interrupted without output")
 	}
 	return nil
 }
+
+func (*cmdRun) handleOutputError(err error) error {
+	if err == nil {
+		return nil
+	}
+	if output.ExportErrorOf(err).HasAuthError() {
+		return errors.New(
+			"authentification to benchttp server failed, " +
+				`please run "benchttp auth login" and restart the benchmark`,
+		)
+	}
+	return err
+}
```

Comment:
Nitpick:
I would have inlined it because when reading `execute` we do not see the explicit full flow and I don't think the extraction provide any benefit on the other hand.

Reply:
Benefits are:
- better readability of the main process (only focus on what is does, not how), which is welcomed here because it orchestrates many things
- decreased complexity